psydac/fem/projectors.py

```diff
@@ -158,3 +160,297 @@ def get_dual_dofs(Vh, f, domain_h, backend_language="python", return_format='ste
         return tilde_f.toarray()
     else:
         return tilde_f
+
+
+#===============================================================================
+class DirichletBoundaryProjector(LinearOperator):
+    """
+    A LinearOperator that applies homogeneous (unless manually given different bcs) Dirichlet boundary conditions.
+
+    Parameters
+    ----------
+    fem_space : psydac.fem.basic.FemSpace
+        fem_space.coeff_space is domain and codomain of this LO. fem_space.kind.name determines the BCs that get applied.
+
+    bcs : Iterable
+        Iterable of sympde.topology.Boundary objects. Necessary only if fem_space.kind.name is undefined.
+
+    space_kind : str | SpaceType
+        Necessary only if fem_space.kind.name is undefined.
+
+    Notes
+    -----
+    See examples/vector_potential_3d.py for a use case of such an operator.
+    
+    """
+    def __init__(self, fem_space, *, bcs=None, space_kind=None):
+
+        assert isinstance(fem_space, FemSpace)
+        assert bcs is None or isinstance(bcs, Iterable)
+        assert space_kind is None or isinstance(space_kind, (str, SpaceType))
+
+        coeff_space    = fem_space.coeff_space
+        self._domain   = coeff_space
+        self._codomain = coeff_space
+
+        if bcs is not None:
+            assert all(isinstance(bc, Boundary) for bc in bcs)
+            self._bcs = tuple(bcs)
+        else:
+            self._bcs = tuple(self._get_bcs(fem_space, space_kind=space_kind))
+
+    @property
+    def domain(self):
+        return self._domain
+    
+    @property
+    def codomain(self):
+        return self._domain
+    
+    @property
+    def dtype(self):
+        return None
+    
+    @property
+    def bcs(self):
+        return self._bcs
+    
+    def tosparse(self):
+        raise NotImplementedError
+    
+    def toarray(self):
+        raise NotImplementedError
+    
+    def transpose(self, conjugate=False):
+        return self
+    
+    def _get_bcs(self, fem_space, space_kind=None):
+        """Returns the correct Dirichlet boundary conditions for the passed fem_space."""
```

Comment:
Please expand this docstring in order to clarify the type of the function parameters.

Also:

```suggestion
    def _get_bcs(self, fem_space, *, space_kind=None):
        """Returns the correct Dirichlet boundary conditions for the passed fem_space."""
```